src/components/Header/index.js

```diff
@@ -1,14 +1,23 @@
 import React from 'react'
-import { View, TextInput, StyleSheet } from 'react-native'
+import { Keyboard, View, TextInput, StyleSheet } from 'react-native'
 
 class Header extends React.Component {
+  handleSubmitEditing = () => {
+    this.props.handleAddItem()
+    Keyboard.dismiss()
```

Comment:
I believe the OS automatically dismisses the keyboard when the input is blurred.